src/classes/UTIL_CustomSettingsFacade.cls

```diff
@@ -50,6 +50,56 @@ public without sharing class UTIL_CustomSettingsFacade {
     public static List<Relationship_Auto_Create__c> relationshipAutocreate;
     public static List<Relationship_Lookup__c> relationshipLookup;
 
+    private static Map<Schema.SObjectType, List<Relationship_Auto_Create__c>> relationshipAutoCreateBySObjectType;
+    private static List<Relationship_Lookup__c> activeRelationshipLookupList;
+
+    /******************************************************************************************************
+    * @description Generates a map of Relationship Auto Create settings for Contact
+    * @return A map containing a list of Relationship Auto Create settings for the Contact object
+    ******************************************************************************************************/
+    public static Map<Schema.SObjectType, List<Relationship_Auto_Create__c>> getAutoCreateSettingsForContact(){
+        if (relationshipAutoCreateBySObjectType != null){
+            return relationshipAutoCreateBySObjectType;
+        }
+
+        if (relationshipAutocreate == null || relationshipAutocreate.isEmpty() == true){
+            relationshipAutocreate = UTIL_CustomSettingsFacade.getAutocreateSettings();
+        }
+
+        List<Relationship_Auto_Create__c> relationshipAutoCreateList = new List<Relationship_Auto_Create__c>();
+
+        for (Relationship_Auto_Create__c relAutoCreate : relationshipAutocreate){
+            if (relAutoCreate.Object__c.equalsIgnoreCase('Contact')){
+                relationshipAutoCreateList.add(relAutoCreate);
+            }
+        }
+
+        relationshipAutoCreateBySObjectType = new Map<Schema.SObjectType, List<Relationship_Auto_Create__c>>();
+        relationshipAutoCreateBySObjectType.put(Contact.SObjectType, relationshipAutoCreateList);
+
+        return relationshipAutoCreateBySObjectType;
+    }
+
+    /******************************************************************************************************
+    * @description Generates a list of active Relationship Lookup mappings.
+    * @return A list containing the current active Relationship Lookup mappings.
+    ******************************************************************************************************/ 
+    public List<Relationship_Lookup__c> getActiveRelationshipLookups(){
+        if (activeRelationshipLookupList != null){
+            return activeRelationshipLookupList;
+        }
+
+        activeRelationshipLookupList = new List<Relationship_Lookup__c>();
+
+        for (Relationship_Lookup__c relLookup : activeRelationshipLookupList){
```

Comment:
This doesn't seem right? activeRelationshipLookupList will always be empty at this line? Should we loop through: this.getReciprocalSettings()?